rasterio/plot.py

```diff
@@ -130,17 +134,30 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
     return ax
 
 
-def plotting_extent(source):
+def plotting_extent(source, transform=None):
     """Returns an extent in the format needed
      for matplotlib's imshow (left, right, bottom, top)
      instead of rasterio's bounds (left, bottom, top, right)
 
     Parameters
     ----------
-    source : raster dataset
+    source : raster dataset or array
+    transform: Affine, required if source is array
     """
-    extent = (source.bounds.left, source.bounds.right,
-              source.bounds.bottom, source.bounds.top)
+    try:
```

Comment:
I like the idea of checking for the attribute as in your code sample. 

In general I'd try to avoid the `isinstance` call since we'd want to support any objects which implement the same interface (e.g. a `Raster` abstraction object or similar)